tasks/__init__.py

```diff
@@ -47,8 +47,8 @@ def _init_event_producer(config):
         return NullProducer()
 
 
-def emit_event(e):
-    producer.send(cfg.event_topic, value=e.encode("utf-8"))
+def emit_event(e, id):
```

Comment:
Same here, I’d name it _key_ and make it optional.

```suggestion
def emit_event(e, key=None):
```